pkg/platform/common/vmlayer/appinst.go

```diff
@@ -316,6 +316,84 @@ func seedDockerSecrets(ctx context.Context, client ssh.Client, clusterInst *edge
 	return nil
 }
 
+func (v *VMPlatform) setupDnsForAppInst(ctx context.Context, clusterInst *edgeproto.ClusterInst, app *edgeproto.App, appInst *edgeproto.AppInst, client ssh.Client, deployment string) error {
+	var proxyOps []proxy.Op
+	var getDnsAction func(svc v1.Service) (*infracommon.DnsSvcAction, error)
+
+	rootLBName := v.VMProperties.GetRootLBNameForCluster(ctx, clusterInst)
+	rootLBIPs, err := v.GetExternalIPFromServerName(ctx, rootLBName, WithCachedIP(true))
+	if err != nil {
+		return err
+	}
+	names, err := k8smgmt.GetKubeNames(clusterInst, app, appInst)
+	if err != nil {
+		return err
+	}
+	sips, err := v.GetIPFromServerName(ctx, v.VMProperties.GetCloudletMexNetwork(), v.GetClusterSubnetName(ctx, clusterInst), GetClusterMasterName(ctx, clusterInst))
+	if err != nil {
+		return err
+	}
+	ops := infracommon.ProxyDnsSecOpts{AddProxy: true, AddDnsAndPatchKubeSvc: true, AddSecurityRules: true}
+	wlParams := infracommon.WhiteListParams{
+		SecGrpName:  infracommon.GetServerSecurityGroupName(rootLBName),
+		ServerName:  rootLBName,
+		Label:       infracommon.GetAppWhitelistRulesLabel(app),
+		AllowedCIDR: infracommon.GetAllowedClientCIDR(),
+		Ports:       appInst.MappedPorts,
+		DestIP:      infracommon.DestIPUnspecified,
+	}
+	proxyConfig := NewProxyConfig(ListenAllIPs, sips, appInst.EnableIpv6)
+
+	// setup common proxy options
+	loopbackIp := infracommon.GetUniqueLoopbackIp(ctx, appInst.MappedPorts)
+	proxyOps = append(proxyOps, proxy.WithDockerNetwork("host"))
+	proxyOps = append(proxyOps, proxy.WithMetricEndpoint(loopbackIp))
+
+	switch deployment {
+	case cloudcommon.DeploymentTypeHelm:
+		fallthrough
+	case cloudcommon.DeploymentTypeKubernetes:
+		patchIPV4 := ""
+		patchIPV6 := ""
+		if v.VMProperties.GetUsesMetalLb() {
+			patchIPV4 = sips.IPV4ExternalAddr()
+			patchIPV6 = sips.IPV6ExternalAddr()
+		}
```

Comment:
Shouldn't this be if not uses metal lb?

Reply:
Thanks! Good catch - I was looking at the logic to make sure I don't mess it up...and still did :)